### ISSUE TITLE
[Feature Request] better UI for apps page

### ISSUE THREAD

**Issue by SelfhostedPro** [SelfhostedPro/Yacht#156]
Need to think of a better way to display the columns, check for updates, and refresh buttons (possibly more if they come up). Any recommendations/ideas?

**Comment by grewebadmin** [comment]
I installed yacht today. I like the way the 'view apps' page currently looks. I guess my only request would be to have a "hide" option for older apps. Overall, the currently layout is intuitive and polished.

**Comment by SelfhostedPro** [comment]
What do you mean as far as hiding older apps? Like sort by newest first by default (only shows 10 apps at a time)?

**Comment by grewebadmin** [comment]
Maybe older was a bad example, minimalists like myself like to hide things I don't go in and out of frequently. Really the filters do a good enough job that hiding isn't necessary, just a thought for people who may have a lot of apps on there.

**Comment by themorlan** [comment]
I also like the design as it is right. What is your concern with?
Maybe replace the arrow down icon with three dots icon to better ilustrate that there are more options available.
![screenshot](https://user-images.githubusercontent.com/48086551/97624089-e5b73280-1a26-11eb-9c58-99c45498cada.png)

**Comment by SelfhostedPro** [comment]
> Maybe older was a bad example, minimalists like myself like to hide things I don't go in and out of frequently. Really the filters do a good enough job that hiding isn't necessary, just a thought for people who may have a lot of apps on there.
> 
> Edit, unrelated to the apps page specifically. Tho I would recommend a "?" icon (like in the picture) that links to: https://yacht.sh/
> Just for quick access to this page without jumping to github first.
> ![My Post](https://user-images.githubusercontent.com/72995533/97050976-54d6e780-154c-11eb-90a2-a194a08ae3fb.jpg)

I will definitely look into adding that. Great suggestion!

The main concern is that if I add more buttons (like how columns is now) it could cause issues in the future with the design. Are there any places in the app that you think could have the UI changed to be better?

**Comment by themorlan** [comment]
> The main concern is that if I add more buttons (like how columns is now) it could cause issues in the future with the design.

I would find it intuitive to have an arrow next to the column headers to filter them:
![screenshot2](https://user-images.githubusercontent.com/48086551/97668758-f9908200-1a82-11eb-8c36-01525038aeec.png)

In the resources tabs I don't like the position of the option dots. I would put them in the first column (but thats personal taste I guess)
![DeepinScreenshot_select-area_20201030073959](https://user-images.githubusercontent.com/48086551/97668981-6d328f00-1a83-11eb-90f4-6da0bec35a9b.png)
![screenshot3](https://user-images.githubusercontent.com/48086551/97668984-6e63bc00-1a83-11eb-99b8-7ffa6f47414d.png)

In general I really like the UI design. It's easy to navigate and most of the time the control elements are where you expect them.

**Comment by grewebadmin** [comment]
Hi themorlan 

Thank you for contacting our support team. A support ticket has now been
opened for your request. You will be notified when a response is made by
email.

Subject: Re: [SelfhostedPro/Yacht] [Feature Request] better UI for apps page
(#156)
Department: IT
Priority: Medium

Ticket message:
> The main concern is that if I add more buttons (like how columns is now)
it could cause issues in the future with the design.
I would find it intuitive to have an arrow next to the column headers to
filter them:
![screenshot2](https://user-images.githubusercontent.com/48086551/97668758-f9908200-1a82-11eb-8c36-01525038aeec.png)
In the resources tabs I don't like the position of the option dots. I would
put them in the first column (but thats personal taste I guess)
![DeepinScreenshot_select-area_20201030073959](https://user-images.githubusercontent.com/48086551/97668981-6d328f00-1a83-11eb-90f4-6da0bec35a9b.png)
![screenshot3](https://user-images.githubusercontent.com/48086551/97668984-6e63bc00-1a83-11eb-99b8-7ffa6f47414d.png)
In general I really like the UI design. It's easy to navigate and most of
the time the control elements are where you expect them.
-- 
You are receiving this because you commented.
Reply to this email directly or view it on GitHub:
https://github.com/SelfhostedPro/Yacht/issues/156#issuecomment-719299725

You can view the ticket on the following link: #5

Kind Regards,


                             
                           

                         

                       
                       
                     

                     
                     

                      

                        
                          

                            GRE - Dashboard

**Comment by HotBlack77** [comment]
I would like to see a lot of what is currently in the Dashboard merged into the apps list. I am thinking the sort of "card" display used in the dashboard, but each card spans the whole width of the page. On each card, it lists the name, container, status, cpu and mem usage, image date, and container creation date and ports, along with icons for each of the actions that can be performed. (I guess this could just be a very wide table, but using cards would allow all the info about each container to listed - there is not that much, just more than one line)
Above the cards there would be some options to sort by the various properties listed in the cards, e.g. mem usage or image date. This motivates the cards being horizontal as they then have a pretty natural sorting direction.

**Comment by SelfhostedPro** [comment]
I'm definitely not opposed to this idea. It's a bit more difficult as there's not a built in component to manage it but I could probably grab some of the functionality from the template lists.

**Comment by ksurl** [comment]
is it possible to have the side menu toggle from mini and fully expanded instead of auto expand on hover? similar behavior to portainer.

also does the roadmap feature supporting docker compose mean a native gui to manage compose files? that is probably my most used feature on portainer, being able to quickly paste a new compose file or edit one and redeploy in yaml format. if the goal is to just use the mounted compose folder and manage that externally with editor of choice, that is a valid design as well but just a bit of an extra step. I guess you could run code-server to do that.

**Comment by SelfhostedPro** [comment]
> is it possible to have the side menu toggle from mini and fully expanded instead of auto expand on hover? similar behavior to portainer.
> 
> also does the roadmap feature supporting docker compose mean a native gui to manage compose files? that is probably my most used feature on portainer, being able to quickly paste a new compose file or edit one and redeploy in yaml format. if the goal is to just use the mounted compose folder and manage that externally with editor of choice, that is a valid design as well but just a bit of an extra step. I guess you could run code-server to do that.

I definitely could have it do that. The main reason I went with expanding is for mobile but it's kind of janky on mobile anyways (definitely looking into fixing that once everything else is done). Is there a more ideal way to present the menu that's not a sidebar?

I am working on an in browser editor. It's currently in the auth_rewrite branch (I'm moving to an auth framework that lets me customize it moreso than my current one too). I provided instructions so one could use codeserver in the meantime. I'm going to recommend using code server regardless as in big compose projects there's a lot of files you need to modify (configurations, etc) and I don't know if I would be able to provide a way to edit all of that cleanly.

**Comment by ksurl** [comment]
actually that's probably better to use code-server since I manage compose file history in git

**Comment by SelfhostedPro** [comment]
Exactly, most of the docker-compose projects I use are cloned from git repos anyways. I may eventually look into having versioning there as well but that's a bit of a ways off. It'll still be nice to have a quick and easy in browser editor too. 

I'll let you know when I've got the sidebar sorted out.

**Comment by ksurl** [comment]
I did my first template via yaml, works well. I just realized projects are the actual compose services and not a template to deploy.

one other feature suggestion, support .yachtignore file for the compose folder. I keep compose files for stuff I'm not currently running. I tried moving them to a subfolder but it looks like yacht still discovers them.

**Comment by ksurl** [comment]
You could make an "edit" button that links to a configurable variable for code-server link. I'm not sure if that's practical or not.

**Comment by SelfhostedPro** [comment]
I'll probably implement some kind of automation to where you can deploy code server with one click but the edit button on the docker-compose page will just be the in browser editor.

**Comment by yangkghjh** [comment]
Some recommendations:

- Add filters to:
 - Status
- Checkbox for listing the apps created by yacht only
- Add Fields
 - Icon
 - Volumes
 - Network mode
 - Restart policy
 - Quick actions to the log and stats
- Add port bind info in port tooltip, like 8080->80
- Relative created/start time like `docker ps`

**Comment by SelfhostedPro** [comment]
@yangkghjh can you be a bit more descriptive with some of these? Yacht doesn't store any data on containers but I can have it add a label to them on deployment (for the last point). May be good to open a separate issue for this.

**Comment by yangkghjh** [comment]
@SelfhostedPro  Fixed the list format issue. Sorry about that.

**Comment by SelfhostedPro** [comment]
> is it possible to have the side menu toggle from mini and fully expanded instead of auto expand on hover? similar behavior to portainer.
> 
> also does the roadmap feature supporting docker compose mean a native gui to manage compose files? that is probably my most used feature on portainer, being able to quickly paste a new compose file or edit one and redeploy in yaml format. if the goal is to just use the mounted compose folder and manage that externally with editor of choice, that is a valid design as well but just a bit of an extra step. I guess you could run code-server to do that.

Docker-compose is now in latest if you want to give it a go. (in-browser editor for the files as well)

**Comment by marcjmiller** [comment]
Is this the place to mention UI bugs?  If not, I can make a new issue.  I'm a React/Typescript dev by day, with a bit of Flask/Python experience if it is something you'd take a MR for.

Currently, I've noticed (at 1080p and 1440p at least), on the App Info page (`http://<YACHT_ADDRESS>/#/apps/<APP_NAME>/info`) the tabs fall behind the main UI element and are smooshed a little, as seen by the highlight here cutting off the bottom:
![image](https://user-images.githubusercontent.com/48929970/109400501-15b00d00-7917-11eb-976d-78b6965b0ac5.png)

The container that is highlighting has some space on the bottom being hidden, as seen here:
![image](https://user-images.githubusercontent.com/48929970/109400582-91aa5500-7917-11eb-928b-6fce8507e208.png)

**Comment by SelfhostedPro** [comment]
@marcjmiller a separate issue will be good for me to tack this specifically. I'm redoing the app page all together this week but I hadn't noticed this specific issue. If you've got any ideas as far as troubleshooting goes, that would be awesome as well as I'm not seeing any errors in the console relating to this.

**Comment by kcoderhtml** [comment]
I would icons to be added to containers like the logo for bitwarden or dashy. I saw this mentioned on the Awesome Open Source youtube channel [https://youtu.be/eTQ2iB-hjkk?t=550](https://youtu.be/eTQ2iB-hjkk?t=550)

**Comment by wickedyoda** [comment]
> I would icons to be added to containers like the logo for bitwarden or dashy. I saw this mentioned on the Awesome Open Source youtube channel [https://youtu.be/eTQ2iB-hjkk?t=550](https://youtu.be/eTQ2iB-hjkk?t=550)

Your commenting to an aged issue, 1 yr old or more. Might want to open a fresh feature request and add all the information and links you can.

**Comment by kcoderhtml** [comment]
Sorry didn't see that. Will do.
